include/kernel.h

```diff
@@ -4883,6 +4883,29 @@ __syscall void k_str_out(char *c, size_t n);
 extern void z_arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 			    void (*fn)(int key, void *data), void *arg);
 
+
+/**
+ * @brief Disable preservation of floating point context information.
+ *
+ * This routine informs the kernel that the specified thread
+ * will no longer be using the floating point registers.
+ *
+ * @warning
+ * Some architectures apply restrictions on how the disabling of floating
+ * point preservation may be requested, see z_arch_float_disable.
+ *
+ * @warning
+ * This routine should only be used to disable floating point support for
+ * a thread that currently has such support enabled.
+ *
+ * @param thread ID of thread.
+ *
+ * @retval 0       On success.
+ * @retval -ENOSYS If the floating point disabling is not implemented.
+ *         -EINVAL If the floating point disabling could not be performed.
+ */
+__syscall int k_float_disable(struct k_thread *thread);
```

Comment:
what about moving k_float_enable as well, right now it is hidden in include/arch/x86/arch.h?

Reply:
I don't think it's ever needed, if you use the FPU and it's not enabled, an exception is triggered and the system just turns it on and retries the instruction.

Reply:
Agreed, though it's possible to imagine architecture/implementions where "FPU enable" is an "allocating" operation that might fail, in which case we want apps to have control over who gets to have one instead of accidentally trapping out of some unrelated thread.  (I mean, sure, it could go on the stack basically everywhere.  But consider pathological cases like an AVX512 system where you need 16*64 = 1024 bytes of space, which is quite a bit larger than our default stacks are.).  Probably not worth an argument, but I can at least imagine someone wanting to do trickery which requires an explicit k_float_enable() call.

Reply:
As @andrewboie mentioned, the x86 architecture does not use the `k_float_enable` call (I recall we had discussed that). So at this very moment we don't need to expose this as a system call.

However, @nashif , @andyross : I am considering to introduce the `k_float_enable` as an API for Arm (still thinking whether we need that). If we decide to do so, I will move it to the public API (for x86 as well).

Until then , let's live with k_float_disable only; what do you think?